common/tests/unittests/test_memory_limit.py

```diff
@@ -0,0 +1,21 @@
+def test_memory_mocking():
+    import psutil
+    from autogluon.common.utils.resource_utils import ResourceManager
+
+    assert psutil.virtual_memory().total == ResourceManager._get_memory_size()
+    assert psutil.virtual_memory().available == ResourceManager._get_available_virtual_mem()
+
+
+def test_custom_memory_soft_limit():
+    import os
+
+    import psutil
+
+    p = psutil.Process()
+    os.environ["AG_MEMORY_LIMIT_IN_GB"] = "48"
```

Comment:
Does this only work if done prior to AutoGluon import?

Reply:
No, this should always work because. Fixed the order.